packages/dev/loaders/src/BVH/bvhLoader.ts

```diff
@@ -0,0 +1,371 @@
+    
+         - data: splitted string array (frame values), values are shift()ed so
+         this should be empty after parsing the whole hierarchy.
+         - frameTime: playback time for this keyframe.
+         - bone: the bone to read frame data from.
+
+         Note: Position data (specifically Z) is flipped to convert coordinate systems.
+    */
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    protected static _readFrameData(data: string[], frameTime: number, bone: BVHNode) {
+        if (bone.type === "ENDSITE")
+            // end sites have no motion data
+            return;
+
+        // add keyframe
+        const keyframe = new BVHKeyFrame();
+        keyframe.time = frameTime;
+        keyframe.position = new Vector3();
+        keyframe.rotation = new Quaternion();
+
+        bone.frames.push(keyframe);
+
+        let pitch = 0,
+            yaw = 0,
+            roll = 0;
+
+        // parse values for each channel in node
+        for (let i = 0; i < bone.channels.length; ++i) {
+            const value = data.shift();
+            if (!value) {
+                continue;
+            }
+
+            switch (bone.channels[i]) {
+                case this.X_POSITION:
+                    keyframe.position.x = parseFloat(value.trim());
+                    break;
+                case this.Y_POSITION:
+                    keyframe.position.y = parseFloat(value.trim());
+                    break;
+                case this.Z_POSITION:
+                    keyframe.position.z = -parseFloat(value.trim()); // Flip Z position
+                    break;
+                case this.X_ROTATION:
+                    pitch = (parseFloat(value.trim()) * Math.PI) / 180;
+                    break;
+                case this.Y_ROTATION:
+                    yaw = (parseFloat(value.trim()) * Math.PI) / 180;
+                    break;
+                case this.Z_ROTATION:
+                    roll = (parseFloat(value.trim()) * Math.PI) / 180;
+                    break;
+                default:
+                    throw new Error("invalid channel type");
+            }
+        }
+
+        if (yaw != 0 || pitch != 0 || roll != 0) {
+            // Create rotation matrix in proper order
+            const rotationMatrix = Matrix.Identity();
+            Matrix.RotationYawPitchRollToRef(yaw, pitch, roll, rotationMatrix);
+            keyframe.rotation.fromRotationMatrix(rotationMatrix);
+        }
+
+        // parse child nodes
+        for (let i = 0; i < bone.children.length; ++i) {
+            BVHLoader._readFrameData(data, frameTime, bone.children[i]);
+        }
+    }
+}
+
+class LoaderContext {
+    animationName: string = "";
+    loopBehavior: number = 0;
+    list: BVHNode[] = [];
+    root: BVHNode = new BVHNode();
+    numFrames: number = 0;
+    frameRate: number = 0;
+    realSkeleton: Skeleton;
+
+    constructor() {
+        this.realSkeleton = null as any; // Will be initialized in readBvh
```

Comment:
Why not just pass the realSkeleton into the constructor so we don't have to have the realSkeleton property be in a bad state temporarily?